src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/menu/AndroidShareMenu.java

```diff
@@ -0,0 +1,226 @@
+package io.github.thebusybiscuit.slimefun4.implementation.items.androids.menu;
+
+import io.github.thebusybiscuit.cscorelib2.inventory.ChestMenu;
+import io.github.thebusybiscuit.cscorelib2.item.CustomItem;
+import io.github.thebusybiscuit.cscorelib2.skull.SkullItem;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import io.github.thebusybiscuit.slimefun4.implementation.items.androids.ProgrammableAndroid;
+import io.github.thebusybiscuit.slimefun4.utils.ChatUtils;
+import io.github.thebusybiscuit.slimefun4.utils.ChestMenuUtils;
+import io.github.thebusybiscuit.slimefun4.utils.HeadTexture;
+import me.mrCookieSlime.Slimefun.api.BlockStorage;
+
+import org.apache.commons.lang.Validate;
+
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.block.Block;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.Nonnull;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * The {@link AndroidShareMenu} is responsibility to handle trusted user using the {@link
+ * ProgrammableAndroid}
+ *
+ * @author StarWishsama
+ * @see ProgrammableAndroid
+ */
+public final class AndroidShareMenu {
+	private static final int startSlot = 9;
+	private static final int perPageDisplay = 45;
+	private static final String key = "share-users";
+
+	private AndroidShareMenu() {}
+
+	/**
+	 * Open a share menu for player.
+	 *
+	 * @param p player
+	 * @param b android
+	 * @param page page
+	 */
+	public static void openShareMenu(@Nonnull Player p, @Nonnull Block b, int page) {
+		Validate.notNull(p, "The player cannot be null!");
+		Validate.notNull(b, "The android block cannot be null!");
+		Validate.isTrue(page >= 0, "The page must be above or equals 0!");
+
+		ChestMenu menu = new ChestMenu(SlimefunPlugin.instance(), SlimefunPlugin.getLocalization().getMessage("android.access-manager.title"));
+
+		menu.setEmptySlotsClickable(false);
+
+		List<String> users = getTrustedUsers(b);
+
+		int pages = Math.max(0, users.size() / 36);
+
+		// Draw background start
+		for (int i = 0; i < 9; i++) {
+			menu.addItem(i, new CustomItem(new ItemStack(Material.GRAY_STAINED_GLASS_PANE), " "));
+			menu.addMenuClickHandler(i, (pl, slot, item, cursor, action) -> false);
+		}
+
+		for (int i = 45; i < 54; i++) {
+			menu.addItem(i, new CustomItem(new ItemStack(Material.GRAY_STAINED_GLASS_PANE), " "));
+			menu.addMenuClickHandler(i, (pl, slot, item, cursor, action) -> false);
+		}
+		// Draw background end
+
+		// Add trusted player slot
+		menu.addItem(0, new CustomItem(HeadTexture.SCRIPT_UP.getAsItemStack(), SlimefunPlugin.getLocalization().getMessage("android.access-manager.menu.add-player-title"), SlimefunPlugin.getLocalization().getMessage("android.access-manager.menu.add-player")));
+		menu.addMenuClickHandler(
```

Comment:
this formatting is crazy

Reply:
looks better then most formatting imo. makes it look way cleaner then those really long lines